Emrald-UI/src/components/forms/StateForm/StateForm.tsx

```diff
@@ -10,19 +10,43 @@ import { v4 as uuidv4 } from 'uuid';
 import MainDetailsForm from '../../forms/MainDetailsForm';
 import { emptyState, useStateContext } from '../../../contexts/StateContext';
 import { useSignal } from '@preact/signals-react';
-import { MainItemTypes, StateType } from '../../../types/ItemTypes';
+import {
+  DiagramType,
+  MainItemTypes,
+  StateEvalValue,
+  StateType,
+} from '../../../types/ItemTypes';
+import {
+  FormControl,
+  FormControlLabel,
+  FormLabel,
+  Radio,
+  RadioGroup,
+} from '@mui/material';
+import { useDiagramContext } from '../../../contexts/DiagramContext';
+import { Diagram } from '../../../types/Diagram';
 
 interface StateFormProps {
   stateData?: State;
+  parentDiagram?: Diagram;
```

Comment:
Might be best to make this required.

```suggestion
 parentDiagram: Diagram;
```